src/Backend/Haskell.idr

```diff
@@ -105,6 +108,60 @@ data Haskell : Type where
   ||| clauses of the form ((arg1, ..., argk), rhs).
   FunDef : Name -> HsType -> List (List HsTerm, HsTerm) -> Haskell
 
+-- Effects -----
+
+||| Environment, contains currently used variables and names
+ENV : Nat -> EFFECT
+ENV n = 'Env ::: STATE (SortedMap String Nat, Vect n (HsType, HsTerm))
+
+||| Monad for the term generator, keeping track of a name supply (in
+||| the form of a map of the number of usages for each name) and a
+||| "semantic" environment consisting of an environment of types, and
+||| terms for decoding/encoding the types.
+||| @ n the size of the environment
+TermGen : (n : Nat) -> Type -> Type
+TermGen n t = Eff t [ENV n, SPECIALIZED HsType, ERR]
+
+toTermGen : Either CompilerError a -> TermGen n a
+toTermGen (Left err) = raise err
+toTermGen (Right val) = pure val
+
+Ord k => Default (SortedMap k v) where
```

Comment:
Is this only used for the initial states below? If so, maybe clearer to just use `empty` directly there? (Since presumably the initial value being empty matters.)

Reply:
Absolutely, I removed the definition and implemented `runLookupM` in `src/Backend/Utils.idr` and I'm using it instead in the haskell backend.